### ISSUE TITLE
use Cirrus CI for some validations 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#8852]
https://cirrus-ci.org/

**Comment by Abhishek-kumar09** [comment]
>use Cirrus CI for some validations

What would be the expected tasks and validations?
I had tried to add it on my own repo: https://github.com/Abhishek-kumar09/checkstyle/blob/master/.cirrus.yml
and put some barebone validations.

I had hard time linking it to Pull requests, Do we have to do some extra stuff integrating ci with pull request.
I had seen cirrus.yml in their own repository, nothing peculiar was written and things were running in their PR section too.

I had successfully ran it on my project. 
![image](https://user-images.githubusercontent.com/48255244/94423440-46dcb380-01a6-11eb-834b-3167549f7a22.png)

**Comment by strkkk** [comment]
@Abhishek-kumar09 
> I had hard time linking it to Pull requests, Do we have to do some extra stuff integrating ci with pull request.

What do you mean? What additional steps are required for proper setup?

**Comment by Abhishek-kumar09** [comment]
@strkkk 
What I had done was:
1. installed cirrus-ci on repo, through their link
2. add cirrus.yml file and wrote two test

3. It was running on the project as a whole (screenshot above)
4. I opened a PR on my repo
But ci-checks were not there, like we have in checkstyle's PR

> What additional steps are required for proper setup?

I guess just installing through their linking and authorizing github repo with cirrus-ci was enough to use it in the project. I did no additional steps except this. It is pretty new so some community and documentation support is also missing, but ci as a whole is good.
